aosp_diff/preliminary/system/core/0001-adb-dbc-tty-add-adb-over-dbc-tty-support.patch

```diff
@@ -0,0 +1,491 @@
+From: Prabhat Chand Pandey <prabhat.chand.pandey@intel.com>
+Date: Sun, 22 Nov 2020 12:59:30 +0530
+Subject: [PATCH] adb: dbc: tty: add adb over dbc tty support
+
+Added adb over dbc tty support in adb framework.
+This implementation is based on Blocking I/O calls.
+Set "setprop persist.vendor.sys.usb.adbover" android
+property to "dbc" to use adb over dbc tty otherwise
+set this property to "dwc" (which is the default value
+of this property) to use adb over normal USB (which need
+USB Device Mode IP to be enabled).
+
+Tracked-On: OAM-94662
+Signed-off-by: Prabhat Chand Pandey <prabhat.chand.pandey@intel.com>
+---
+ adb/adb.h                    |   6 ++
+ adb/client/commandline.cpp   |  11 ++--
+ adb/client/transport_usb.cpp |   6 +-
+ adb/client/usb_linux.cpp     |  41 +++++++++++++
+ adb/daemon/main.cpp          |   2 +-
+ adb/daemon/usb.cpp           |  21 ++++++-
+ adb/daemon/usb_ffs.cpp       |  67 +++++++++++++++++++++
+ adb/daemon/usb_ffs.h         |   1 +
+ adb/transport.cpp            | 111 +++++++++++++++++++++++++++++++++++
+ adb/transport.h              |  25 ++++++++
+ 10 files changed, 283 insertions(+), 8 deletions(-)
+
+diff --git a/adb/adb.h b/adb/adb.h
+index ce12a55f9..1d8d3d452 100644
+--- a/adb/adb.h
++++ b/adb/adb.h
+@@ -196,6 +196,10 @@ void put_apacket(apacket* p);
+ #define ADB_SUBCLASS 0x42
+ #define ADB_PROTOCOL 0x1
+ 
++#define ADB_DBC_CLASS 0xDC
++#define ADB_DBC_SUBCLASS 0x2
++#define ADB_DBC_PROTOCOL 0x1
++
+ void local_init(const std::string& addr);
+ bool local_connect(int port);
+ int local_connect_arbitrary_ports(int console_port, int adb_port, std::string* error);
+@@ -216,6 +220,8 @@ extern const char* adb_device_banner;
+ #define USB_FFS_ADB_EP0 USB_FFS_ADB_EP(ep0)
+ #define USB_FFS_ADB_OUT USB_FFS_ADB_EP(ep1)
+ #define USB_FFS_ADB_IN USB_FFS_ADB_EP(ep2)
++
++#define USB_DBC_ADB_PATH  "/dev/ttyDBC0"
+ #endif
+ 
+ enum class HostRequestResult {
+diff --git a/adb/client/commandline.cpp b/adb/client/commandline.cpp
+index ad4e21c4d..f98c8b3fb 100644
+--- a/adb/client/commandline.cpp
++++ b/adb/client/commandline.cpp
+@@ -1118,7 +1118,7 @@ static bool adb_root(const char* command) {
+     }
+ 
+     // Figure out whether we actually did anything.
+-    char buf[256];
++/*    char buf[256];
+     char* cur = buf;
+     ssize_t bytes_left = sizeof(buf);
+     while (bytes_left > 0) {
+@@ -1143,15 +1143,16 @@ static bool adb_root(const char* command) {
+     if (cur != buf && strstr(buf, "restarting") == nullptr) {
+         return true;
+     }
+-
++*/
+     // Wait for the device to go away.
+     TransportType previous_type;
+     const char* previous_serial;
+     TransportId previous_id;
+     adb_get_transport(&previous_type, &previous_serial, &previous_id);
+ 
+-    adb_set_transport(kTransportAny, nullptr, transport_id);
+-    wait_for_device("wait-for-disconnect");
++    // adb_set_transport(kTransportAny, nullptr, transport_id);
++    // wait_for_device("wait-for-disconnect");
```

Comment:
Add a comment on why the above lines are commented out

Reply:
ACK